R/create_protocol.R

```diff
@@ -0,0 +1,233 @@
+#' @param revision A semantic version number of the form major.minor.patch. For development versions a fourth component is appended starting from .9000. The default is 0.0.0.9000 and should normally not be changed.
+#' @param projectname A character string giving the project name. Best to use an acronym for this.
+#' @param theme A character string equal to generiek (default), water, lucht, bodem, vegetatie of soorten.
+#'
+#' @return A new subfolder beneath src with an Rmarkdown file
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom stringr str_replace_all str_extract
+#' @importFrom assertthat assert_that is.string is.date
+#' @importFrom rmarkdown draft render
+#' @importFrom purrr map_chr
+#' @importFrom whisker whisker.render
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' create_sfp()
+#'}
+create_sfp <- function(
+  title = "Titel",
+  subtitle = "",
+  short_title = "korte titel",
+  authors = "voornaam1 naam1, voornaam2 naam2, ...",
+  date = Sys.Date(),
+  reviewers = "voornaam1 naam1, voornaam2 naam2, ...",
+  file_manager = "voornaam naam",
+  revision = "0.0.0.9000",
+  projectname = "",
+  theme = c("generiek", "water", "lucht", "bodem", "vegetatie", "soorten")) {
+
+  # check parameters
+  assert_that(is.string(title))
+  assert_that(is.string(subtitle))
+  assert_that(is.string(short_title), nchar(short_title) <= 20)
+  assert_that(is.date(as.Date(date)))
+  assert_that(is.string(authors))
+  assert_that(is.string(reviewers))
+  assert_that(is.string(file_manager))
+  assert_that(is.string(revision))
+  assert_that(is.string(projectname))
+  theme <- match.arg(theme)
+
+  # create protocol name
+
+  protocol_type <- "sfp"
+
+  protocol_leading_number <- switch(theme,
+                                 generiek = "0",
+                                 water = "1",
+                                 bodem = "2",
+                                 lucht = "3",
+                                 vegetatie = "4",
+                                 soorten = "5"
+  )
+
+  all_numbers <- get_protocolnumbers(protocol_type = protocol_type)
+
+  if (length(all_numbers) == 0) {
+    protocol_trailing_number <- "01"
+  } else {
+    numbers <- str_subset(all_numbers, paste0("^", protocol_leading_number))
+    protocol_trailing_number <- max(
+      as.integer(
+        str_extract(numbers, "\\d{2}$"))
+    ) + 1
+    protocol_trailing_number <- formatC(protocol_trailing_number,
+                                   width = 2, format = "d", flag = "0")
+  }
+
+  protocol_number <- paste0(protocol_leading_number,
+                            protocol_trailing_number)
+
+  short_title <- tolower(short_title)
+  short_title <- str_replace_all(short_title, " ", "-")
+  short_titles <- get_short_titles(protocol_type = protocol_type)
+  assert_that(!(short_title %in% short_titles),
+              msg = "De opgegeven korte titel bestaat al. Geef een korte titel die nog niet bestaat.")
```

Comment:
use an English error message

Reply:
Shoot, missed that one. BTW: although the functions will be in English, the protocols will be in Dutch language, hence some friction (title = "Mijn Nederlandstalige titel") will remain. I started out doing everything in Dutch but switched to English due to comments on that decision (see other repo).